test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -531,7 +531,7 @@ upgrade_to_latest=false
 # The 2 arrays should be adjusted over time when newer branches are created,
 # or older branches are EOL.
 newer_release_branches=(develop mongodb-5.0 mongodb-4.4 mongodb-4.2)
-older_release_branches=(mongodb-4.2 mongodb-4.0 mongodb-3.6)
+older_release_branches=(mongodb-4.2)
```

Comment:
@lukech I'm not sure whether I should delete 4.0 here in case this list is used to test upgrading from 4.0?

Reply:
We should keep `mongodb-4.0` for backward compatibility verification.

Reply:
Okay, that makes sense.